OptimizelySDK/Odp/OdpSegmentApiManager.cs

```diff
@@ -0,0 +1,260 @@
+﻿/* 
+ * Copyright 2022 Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Newtonsoft.Json;
+using OptimizelySDK.AudienceConditions;
+using OptimizelySDK.ErrorHandler;
+using OptimizelySDK.Logger;
+using OptimizelySDK.Odp.Entity;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OptimizelySDK.Odp
+{
+    /// <summary>
+    /// Manager for communicating with the Optimizely Data Platform GraphQL endpoint
+    /// </summary>
+    public class OdpSegmentApiManager : IOdpSegmentApiManager
+    {
+        /// <summary>
+        /// Standard message for audience querying fetch errors
+        /// </summary>
+        private const string AUDIENCE_FETCH_FAILURE_MESSAGE = "Audience segments fetch failed";
+
+        /// <summary>
+        /// Error handler used to record errors
+        /// </summary>
+        private readonly IErrorHandler _errorHandler;
+
+        /// <summary>
+        /// Logger used to record messages that occur within the ODP client 
+        /// </summary>
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Http client used for handling requests and responses over HTTP
+        /// </summary>
+        private readonly HttpClient _httpClient;
+
+        /// <summary>
+        /// Manager for communicating with the Optimizely Data Platform (ODP) GraphQL endpoint
+        /// </summary>
+        /// <param name="logger">Collect and record events to log</param>
+        /// <param name="errorHandler">Handler to record exceptions</param>
+        /// <param name="httpClient">HttpClient to use to send queries to ODP</param>
+        public OdpSegmentApiManager(ILogger logger = null, IErrorHandler errorHandler = null,
+            HttpClient httpClient = null
+        )
+        {
+            _logger = logger ?? new NoOpLogger();
+            _errorHandler = errorHandler ?? new NoOpErrorHandler();
+            _httpClient = httpClient ?? new HttpClient();
```

Comment:
My recommendation is to create an static method that will return `HttpClient`, should be use across all our SDKs. May be not now but add a ticket in jira to have a single source to return HttpClient in case HttpClient is not provided. 

https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netframework-4.7#examples
```
HttpClient is intended to be instantiated once per application, rather than per-use.
```

Reply:
New Jira issue created to address in a later effort.